docs/usage/customization-guide.md

```diff
@@ -20,13 +20,96 @@ sort: 7
 NFD provides multiple extension points for vendor and application specific
 labeling:
 
+- [`NodeFeature`](#nodefeature-custom-resource) (EXPERIMENTAL) objects can be
+  used to communicate "raw" node features and node labeling requests to
+  nfd-master.
 - [`NodeFeatureRule`](#nodefeaturerule-custom-resource) objects provide a way to
   deploy custom labeling rules via the Kubernetes API
 - [`local`](#local-feature-source) feature source of nfd-worker creates
   labels by executing hooks and reading files
 - [`custom`](#custom-feature-source) feature source of nfd-worker creates
   labels based on user-specified rules
 
+## NodeFeature custom resource
+
+**EXPERIMENTAL**
+NodeFeature objects provide a way for 3rd party extensions to advertise custom
+features, both as "raw" features that serve as input to
+[NodeFeatureRule](#nodefeaturerule-custom-resource) objects and as feature
+labels directly.
+
+Note that RBAC rules must be created for each extension for them to be able to
+create and manipulate NodeFeature objects in their namespace.
+
+Support for NodeFeature CRD API is enabled with the `-enable-nodefeature-api`
```

Comment:
Speaking of which, we should enable this via WorkerConfig, not to have to set the flag if it is already running

Reply:
I specifically didn't want to have that for now. I had some reasoning in the commit message:
https://github.com/kubernetes-sigs/node-feature-discovery/commit/237494463b4b12e6a28d7207b21ccc0350dbc327

We can add that later if needed, but after careful consideration, I'd say